x/oracle/abci.go

```diff
@@ -0,0 +1,31 @@
+package oracle
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/medibloc/panacea-core/v2/x/oracle/keeper"
+	"github.com/medibloc/panacea-core/v2/x/oracle/types"
+)
+
+func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
+	handlerOracleUpgrade(ctx, keeper)
+}
+
+func handlerOracleUpgrade(ctx sdk.Context, keeper keeper.Keeper) {
+	upgradeInfo, err := keeper.GetOracleUpgradeInfo(ctx)
+	if err != nil {
+		if errors.Is(err, types.ErrOracleUpgradeInfoNotFound) {
+			return
+		} else {
+			panic(err)
+		}
+	}
+
+	if upgradeInfo.ShouldExecute(ctx) {
+		if err := keeper.ApplyUpgrade(ctx, upgradeInfo); err != nil {
+			panic(err)
+		}
+		keeper.RemoveOracleUpgradeInfo(ctx)
```

Comment:
How about not deleting the upgrade info?
When a new upgrade proposal passes, it will be overwritten anyway.

The reason I'm talking about this is because of the following case:

If the oracles didn't upgrade until the target height, they had to re-request `register-oracle` transaction again to be shared the oracle private key in the new binary(upgrade version). 
However, at this time, the oracles will be able to enter the commission rate, which overwrites the existing oracle commission rate. 
In other words, this case should also be done through `upgrade-oracle`, and to do so, it seems that upgrade info should be left. upgrade info will mean upcoming or recent upgrade information.

--
FYI) Plus, in `register-oracle`, the check that if the oracle has already been set or not should be added.

Reply:
That's a good idea.
Then, how about `OracleKeyMigration` rather than a name `OracleUpgrade`?
OracleKeyMigration works only when it has the same value as uniqueID of `OracleUpgradeInfo`.

Reply:
Oh, please forget the name change. I'll think about it more.

Reply:
Okay, I understand the problem. 
If the oracles didn't upgrade until the target height, oracle operator have to use `upgrade-oracle` because the oracle is already in `Oracle` store.
For now, I think @H4NLee 's suggestion is the best way.

Reply:
If you change this flow, please left comment :)

Reply:
I fixed it in [8e1a3d8](https://github.com/medibloc/panacea-core/pull/578/commits/8e1a3d8338c0d7fdc61e0a27556102f708c5fe00).